### ISSUE TITLE
Quick question (tracks only play for 20 secs and do not start at the beginning)

### ISSUE THREAD

**Issue by kyeesa** [kaaes/spotify-web-api-android#110]
Hi, 

Am a student at the university of Nottingham. My dissertation is woking on a music app using some of your work. So firstly **thank you!** My app will hopefully log in to spotty using a premium account, then search for tracks. I want to add selected tracks to a playlist. From this playlist I want to then play these songs. My app's main aim is to have a voting feature that will decide weather to keep the track currently playing in the playlist or not. 

Basically I have downloaded your work but I notice that when searching for a track and then selecting to play it... it only plays 20 seconds of the music and it starts in the middle not the beginning. Why does it do this and how can I change it?

Thank you in advance! **any help would be amazing!**

Kye Esa :-)

**Comment by luwo1701** [comment]
Pretty sure this is because you haven't signed in as a premium user.

**Comment by Servus93** [comment]
In **SearchPresenter.java** change
`    @Override
    public void selectTrack(Track item) {
        String previewUrl = item.preview_url;
//...
}
`
to
`    @Override
    public void selectTrack(Track item) {
        String previewUrl = item.uri;
//...
}
`

You only get a preview of the selected Track, Change the above code to get the path to the whole song.

**Comment by kaaes** [comment]
Web API only gives you 30 seconds previews of the songs. If you want to play full tracks you need to have a premium Spotify account and use [Spotify Android SDK](https://developer.spotify.com/technologies/spotify-android-sdk/).
